locations/spiders/brico_io_it.py

```diff
@@ -1,38 +1,44 @@
+import collections
+
 from scrapy import Spider
 
-from locations.hours import DAYS_IT, OpeningHours, sanitise_day
-from locations.items import Feature
+from locations.dict_parser import DictParser
+from locations.hours import DAYS, OpeningHours
 
 
 class Brico_ioITSpider(Spider):
     name = "brico_io_it"
-    item_attributes = {"brand_wikidata": "Q15965705"}
-    start_urls = ["https://www.bricoio.it/api/organa/stores.aspx"]
+    item_attributes = {"brand": "Brico io", "brand_wikidata": "Q15965705"}
+    allowed_domains = ["www.bricoio.it"]
+    start_urls = ["https://www.bricoio.it/it/it/brit/storelocator/getallstores"]
 
-    def parse(self, response, **kwargs):
+    def parse(self, response):
         for location in response.json():
-            if not location["Attivo"]:
+            if not location["isActive"]:
                 continue
 
-            item = Feature()
-            item["ref"] = location["Codice"]
-            item["name"] = location["Nome"]
-            item["street_address"] = location["Indirizzo"]["Descrizione"]
-            item["city"] = location["Indirizzo"]["Localita"]
-            item["state"] = location["Indirizzo"]["Provincia"]
-            item["postcode"] = location["Indirizzo"]["CodicePostale"]
-            item["country"] = location["Indirizzo"]["Nazione"]
-            item["phone"] = location["Telefono"].replace("/", "")
-            item["email"] = location["Email"]
-            item["lat"] = location["Coordinate"]["Latitudine"]
-            item["lon"] = location["Coordinate"]["Longitudine"]
-            item["website"] = f'https://www.bricoio.it{location["URL"]}'
+            item = DictParser.parse(location)
+            item["ref"] = location["code"]
+            item["name"] = location["alias"]
+            address_fields = {k: v for k, v in DictParser.parse(location["address"]).items() if v}
+            print(address_fields)
+            item["street_address"] = " ".join(
+                filter(None, [location["address"]["street1"], location["address"]["street2"]])
+            )
+            item.update(address_fields)
```

Comment:
Why do we have to call DictParser a second time on the address?

Just looked at the data, not ran it, but I think this should work:

```suggestion
            item["street_address"] = ", ".join(
                filter(None, [location["address"]["street1"], location["address"]["street2"]])
            )
            item["country"] = location["address"]["idCountry"]
```

Reply:
Thanks, fixed. I hadn't even realised I left a print() in there as well!